hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/DefaultS3ClientFactory.java

```diff
@@ -91,6 +95,43 @@ public S3Client createS3Client(
         .build();
   }
 
+  @Override
+  public S3Client createS3EncryptionClient(final S3AsyncClient s3AsyncClient,
+      final S3Client s3Client, final KmsKeyring keyring) {
+
+    return S3EncryptionClient.builder()
+        .wrappedAsyncClient(s3AsyncClient)
+        .wrappedClient(s3Client)
+        .enableLegacyUnauthenticatedModes(true)
+        .keyring(keyring)
```

Comment:
we've had some interest in supporting customer supplied keys too...this would be the time to add it -or at least leave the points for it open. in particular, that third argument should be a structure which contains the keyring and some enum of option to use